logstash-core/lib/logstash/instrument/collector.rb

```diff
@@ -19,7 +19,7 @@ class Collector
     include Observable
     include Singleton
 
-    SNAPSHOT_ROTATION_TIME_SECS = 10 # seconds
+    SNAPSHOT_ROTATION_TIME_SECS = 1 # seconds
```

Comment:
It should be in fact exposed to the user when the configuration is done, so right now I think using 1 is not a real issue until I have benchmarked the code.